rlclientlib/logger/logger_facade.cc

```diff
@@ -124,9 +124,9 @@ int interaction_logger_facade::log(const char* episode_id, const char* previous_
 
       std::string event_id(response.get_event_id());
       std::string model_id(response.get_model_id());
-      std::string previous_id_str(previous_id);
+      std::string previous_id_str(previous_id ? previous_id : "");
```

Comment:
I'm not 100% sure this is the desired behavior? rl_sim sends nullptr for the previous id, which crashed here, but is an empty string the right replacement?

Reply:
I believe that nullptr should be ok here and flatbuffers will translate it to an empty string on deserialization

Reply:
Previously this leads to `string(nullptr)`, but it is undefined behavior, and looks like some compiler changes make it runtime error now and will crash. So this change should be good.